### PR TITLE
Test combined infra metrics

### PR DESCRIPTION
**Pull request by arnout**
A number of fixes are required along the way.

### CODE REVIEW

File: tests/test_flows.sh [hunk 1]
```diff
@@ -84,6 +84,19 @@ send_CAPI_command() {
...
 
+send_CAPI_1905() {
+    container="$1"; shift
+    dest="$1"; shift
+    messagetype="$1"; shift
+    command="DEV_SEND_1905,DestALid,${dest},MessageTypeValue,${messagetype}"
+    for arg; do
+        command="$command,$arg"
+    done
+    capi_command_mid=""
```
Comment (tomereli) [inline tests/test_flows.sh:95]: Why is this needed? you set this 2 lines below

File: tests/test_flows.sh [hunk 2]
```diff
@@ -207,35 +220,63 @@ test_client_capability_query() {
...
     sleep 1
     dbg "Confirming ap capability query has been received on agent"
     check_log ${REPEATER1} agent_wlan0 "AP_CAPABILITY_QUERY_MESSAGE"
     return $check_error
 }
+
 test_combined_infra_metrics() {
-    err "combined_infra_metrics not implemented yet."
-    return 0
+    status "test combined infra metrics"
```
Comment (tomereli) [inline tests/test_flows.sh]: This has to be in a separate commit

**Review by tomereli — CHANGES_REQUESTED** [review]
Looking good so far, but the test MUST make sure the agents actually sends the link metrics and check that the controller adds these as TLVs to the combined infrastructure message. This is the bug that is fixed by #669 so we have to be able to catch these kind of bugs in the test.

**Comment by tomereli** [comment]
> Looking good so far, but the test MUST make sure the agents actually sends the link metrics and check that the controller adds these as TLVs to the combined infrastructure message. This is the bug that is fixed by #669 so we have to be able to catch these kind of bugs in the test.

@YuriyMasechko I don't see that this is taken care of by this PR. The problem is that when the UCC requests to send an empty combined infrastructure metrics like you did here:
```
# Trigger combined infra metrics
dbg "Send Combined infrastructure metrics message to agent 1"
check send_CAPI_1905 ${GATEWAY} $mac_agent1 0x8013
```

The UCC listener should send the preset certification CMDU. If there is no such CMDU (which was the bug solved in my other PR), then the UCC listener will send an *empty* 1905 CMDU with message type 0x8013 - the test failed in the certification since the CMDU was empty, while it should have contained TLVs.

In this case, I expect the test to simulate the same thing and actually verify that the received CMDU in the agent contains the TLVs we expect.
This is why I suggested to add dummy link metrics for now in the agent, so we can verify that the combined infrastructure metrics include these same values and is not just an empty CMDU which is a bug.

**Review by tomereli — CHANGES_REQUESTED** [review]
> Looking good so far, but the test MUST make sure the agents actually sends the link metrics and check that the controller adds these as TLVs to the combined infrastructure message. This is the bug that is fixed by #669 so we have to be able to catch these kind of bugs in the test.

@YuriyMasechko I don't see that this is taken care of by this PR. The problem is that when the UCC requests to send an empty combined infrastructure metrics like you did here:
```
# Trigger combined infra metrics
dbg "Send Combined infrastructure metrics message to agent 1"
check send_CAPI_1905 ${GATEWAY} $mac_agent1 0x8013
```

The UCC listener should send the preset certification CMDU. If there is no such CMDU (which was the bug solved in my other PR), then the UCC listener will send an *empty* 1905 CMDU with message type 0x8013 - the test failed in the certification since the CMDU was empty, while it should have contained TLVs.

In this case, I expect the test to simulate the same thing and actually verify that the received CMDU in the agent contains the TLVs we expect.
This is why I suggested to add dummy link metrics for now in the agent, so we can verify that the combined infrastructure metrics include these same values and is not just an empty CMDU which is a bug.

**Comment by YuriyMasechko** [comment]
> > Looking good so far, but the test MUST make sure the agents actually sends the link metrics and check that the controller adds these as TLVs to the combined infrastructure message. This is the bug that is fixed by #669 so we have to be able to catch these kind of bugs in the test.
> 
> @YuriyMasechko I don't see that this is taken care of by this PR. The problem is that when the UCC requests to send an empty combined infrastructure metrics like you did here:
> 
> ```
> # Trigger combined infra metrics
> dbg "Send Combined infrastructure metrics message to agent 1"
> check send_CAPI_1905 ${GATEWAY} $mac_agent1 0x8013
> ```
> 
> The UCC listener should send the preset certification CMDU. If there is no such CMDU (which was the bug solved in my other PR), then the UCC listener will send an _empty_ 1905 CMDU with message type 0x8013 - the test failed in the certification since the CMDU was empty, while it should have contained TLVs.
> 
> In this case, I expect the test to simulate the same thing and actually verify that the received CMDU in the agent contains the TLVs we expect.
> This is why I suggested to add dummy link metrics for now in the agent, so we can verify that the combined infrastructure metrics include these same values and is not just an empty CMDU which is a bug.

Added check for Transmitter/Receiver link metrics TLV inside handle_1905_combined_infrastructure_metrics() and test_flows.

**Review by tomereli — APPROVED** [review]

File: tests/test_flows.sh (hunk 1 above)
Comment (arnout) [reply tests/test_flows.sh:95]: (For future reference) It is needed because of the `return $?` in the following line, to make sure the mid is not set if the command fails.
